File: client/app/src/main/java/pt/ulisboa/tecnico/cmov/pharmacist/medicine/dto/AddMedicineDto.kt
package pt.ulisboa.tecnico.cmov.pharmacist.medicine.dto

import kotlinx.serialization.Serializable

@Serializable
data class AddMedicineDto(
    val name : String,
    val pharmacyName : String,
    val quantity : Int,
    val purpose : String,
    val photoPath : String
)

File: server/src/medicine/dto/add-medicine.dto.ts
import { IsNotEmpty, IsString, IsOptional, IsNumber } from "class-validator";

export class AddMedicineDto{
    @IsNotEmpty()
    @IsString()
    readonly name: string;

    @IsNotEmpty()
    @IsString()
    readonly pharmacyName : string;

    @IsNotEmpty()
    readonly quantity: number;

    @IsNotEmpty()
    @IsString()
    readonly purpose: string;

    @IsOptional()
    @IsString()
    readonly photoPath?: string;
}
